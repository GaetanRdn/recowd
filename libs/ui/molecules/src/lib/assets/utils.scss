@function map-get-deep($map, $level, $subLevel) {
  @return map-get($map, $level, $subLevel);
}

@mixin use-typography($typography) {
  @each $key in font-size, font-family, font-weight, line-height {
    #{$key}: map-get($typography, $key);
  }
}
