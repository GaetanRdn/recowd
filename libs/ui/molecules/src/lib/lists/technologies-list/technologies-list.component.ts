import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ListModule } from '@recowd/ui-atoms';
import { TranslateModule } from '@ngx-translate/core';

@Component({
  selector: 'rc-technologies-list',
  standalone: true,
  imports: [CommonModule, ListModule, TranslateModule],
  template: ` <rc-list horizontal>
    <rc-list-title>{{ 'technologies' | translate }}</rc-list-title>
    <rc-list-item *ngFor="let techno of technologies">{{
      techno
    }}</rc-list-item>
  </rc-list>`,
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class TechnologiesListComponent {
  @Input() public technologies: string[] = [];
}
