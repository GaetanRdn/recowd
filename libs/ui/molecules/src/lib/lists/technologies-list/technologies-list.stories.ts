import { Meta, moduleMetadata, Story } from '@storybook/angular';
import { TechnologiesListComponent } from './technologies-list.component';

export default {
  title: 'molecules/lists/technologies-list',
  decorators: [
    moduleMetadata({
      imports: [TechnologiesListComponent],
    }),
  ],
} as Meta<TechnologiesListComponent>;

const Template: Story<TechnologiesListComponent> = (
  args: TechnologiesListComponent
) => ({
  props: {
    roles: args.technologies,
  },
  template: `<rc-technologies-list [technologies]="roles"></rc-technologies-list>`,
});

export const Default = Template.bind({});
Default.args = {
  technologies: ['Angular', 'Storybook', 'Nx'],
};
