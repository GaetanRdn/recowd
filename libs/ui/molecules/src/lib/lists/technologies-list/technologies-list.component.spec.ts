import { TestBed, waitForAsync } from '@angular/core/testing';
import { TechnologiesListComponent } from './technologies-list.component';
import { TemplateLookup } from '@recowd/test/utils';
import { Component } from '@angular/core';
import { TranslateModule } from '@ngx-translate/core';

describe('TechnologiesListComponent', () => {
  let templateLookup: TemplateLookup<HostComponent>;

  beforeEach(waitForAsync(() => {
    TestBed.configureTestingModule({
      declarations: [HostComponent],
      imports: [TechnologiesListComponent, TranslateModule.forRoot()],
    }).compileComponents();

    templateLookup = new TemplateLookup(HostComponent);
    templateLookup.detectChanges();
  }));

  it('should create empty', () =>
    expect(templateLookup.firstChildElement).toMatchSnapshot());

  it('should create with technos', () => {
    templateLookup.hostComponent.technologies = ['Angular', 'Jest'];

    templateLookup.detectChanges();

    expect(templateLookup.firstChildElement).toMatchSnapshot();
  });
});

@Component({
  template: ` <rc-technologies-list
    [technologies]="technologies"
  ></rc-technologies-list>`,
})
class HostComponent {
  public technologies: TechnologiesListComponent['technologies'] = [];
}
