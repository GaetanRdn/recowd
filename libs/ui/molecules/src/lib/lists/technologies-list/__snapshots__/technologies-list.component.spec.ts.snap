// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TechnologiesListComponent should create empty 1`] = `
<rc-technologies-list>
  <rc-list
    class="rc-horizontal"
    horizontal=""
    role="list"
  >
    <rc-list-title
      class="rc-list-title"
    >
      technologies
    </rc-list-title>
    <span
      class="rc-list-items-wrapper"
    >
      
    </span>
  </rc-list>
</rc-technologies-list>
`;

exports[`TechnologiesListComponent should create with technos 1`] = `
<rc-technologies-list>
  <rc-list
    class="rc-horizontal"
    horizontal=""
    role="list"
  >
    <rc-list-title
      class="rc-list-title"
    >
      technologies
    </rc-list-title>
    <span
      class="rc-list-items-wrapper"
    >
      <rc-list-item
        class="rc-list-item"
        role="list-item"
      >
        Angular
      </rc-list-item>
      <rc-list-item
        class="rc-list-item"
        role="list-item"
      >
        Jest
      </rc-list-item>
      
    </span>
  </rc-list>
</rc-technologies-list>
`;
